src/server/game/Achievements/AchievementMgr.cpp

```diff
@@ -65,7 +65,7 @@ bool AchievementMgr::CanUpdateCriteriaTree(Criteria const* criteria, CriteriaTre
 {
     AchievementEntry const* achievement = tree->Achievement;
     if (!achievement)
-        return false;
+        return true;
```

Comment:
Yea a new manager for that would be the best thing. If I have time I will have a look at it but I think it's best to leave this work to you and other more experienced people than me :)
Thanks for your time.